eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpJsonItemReader.java

```diff
@@ -0,0 +1,95 @@
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemReader;
+import uk.ac.ebi.eva.accession.dbsnp.model.DbsnpJson;
+import uk.ac.ebi.eva.accession.dbsnp.parameters.InputParameters;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Item reader for Dbsnp Json import job
+ */
+public class DbsnpJsonItemReader implements ItemReader<DbsnpJson> {
+
+    private static final Logger logger = LoggerFactory.getLogger(DbsnpJsonItemReader.class);
+
+    private InputParameters parameters;
+
+    public DbsnpJsonItemReader(InputParameters parameters) {
+        this.parameters = parameters;
+    }
+
+    @Override
+    public DbsnpJson read() throws Exception {
+        JsonFactory jsonFactory = new JsonFactory();
+        JsonParser jp = jsonFactory.createJsonParser(new File(parameters.getDbsnp()));
+        jp.setCodec(new ObjectMapper());
+        JsonNode jsonNode = jp.readValueAsTree();
+        return parseDbsnpJsonPlacements(jsonNode);
+    }
+
+    /**
+     * Parses the JSON as per https://github.com/ncbi/dbsnp/blob/master/tutorials/hadoop_json_placement.py
+     * @param JsonNode jsonRootNode
+     * @return parsed DbsnpJson object
+     */
+    private DbsnpJson parseDbsnpJsonPlacements(JsonNode jsonRootNode) throws IOException {
+
+        DbsnpJson dbsnpJson = new DbsnpJson();
+        dbsnpJson.setClusteredVariantAccession(jsonRootNode.path("refsnp_id").asLong());
+
+        //TODO for now skip handling of case when primary_snapshot_data isn't present
+        //TODO decide for that case between an exception with get(), or with an if size = 0 conditional with path()
+        JsonNode infoNode = jsonRootNode.path("primary_snapshot_data").path("placements_with_allele");
+
+        for(JsonNode alleleInfo : infoNode) {
+
+            boolean isPtlp = alleleInfo.path("is_ptlp").asBoolean();
+            JsonNode assemblyInfo = alleleInfo.path("placement_annot").path("seq_id_traits_by_assembly");
+            if(assemblyInfo.size() == 0 || !isPtlp) {
+                continue;
+            }
+
+            dbsnpJson.setProjectAccession(assemblyInfo.get(0).path("assembly_name").asText());
+
+            for (JsonNode alleleAndHgvs : alleleInfo.path("alleles")) {
+                JsonNode spdi = alleleAndHgvs.path("allele").path("spdi");
+                String insertedSequence = spdi.path("inserted_sequence").asText();
+                String deletedSequence = spdi.path("deleted_sequence").asText();
+                if(!insertedSequence.equals(deletedSequence)) {
+                    dbsnpJson.setReferenceAllele(deletedSequence);
+                    dbsnpJson.setAlternateAllele(insertedSequence);
+                    dbsnpJson.setStart(spdi.path("position").asLong());
+                    dbsnpJson.setContig(spdi.path("seq_id").asText());
```

Comment:
Have you consulted Ensembl's code to check if this is how they parse the variants? A good exercise at this time would be to run Ensembl's code and this pipeline in parallel for a representative sample of variants (that should include multi-allelic variants) and verify if the results match.

Reply:
I would need to setup their code on my machine for that. Going through their Perl parsing logic as of now.

So if I understand correctly, actionable steps: 
1. Isolate 50 JSON fields (check if it has multi-allelic variants defined, if not add more JSON attrs)
2. The above would require to modify the reader to accept a JSON array, instead of single JSON. 
3. Setup Ensembl's code on my machine and run the two pipelines in parallel.

Reply:
You don't have to add more JSON attributes. You just have to look for multiple occurrences of the "allele" attribute within **primary_snapshot_data -> placements_with_allele ->alleles**. See 
[rs1413947121.zip](https://github.com/EBIvariation/eva-accession/files/3300318/rs1413947121.zip) and the [NCBI link](https://www.ncbi.nlm.nih.gov/snp/rs1413947121)

Reply:
So, because an iVariant can have only 1 alternate [variation-commons/iVariant](https://github.com/EBIvariation/variation-commons/blob/fda0d8bbf44f029ab7ef1f756f31964b3bc9e2c5/variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/IVariant.java)
What that means is to call itemProcessor for every loop iteration.